Tensile/TensileCreateLibrary.py

```diff
@@ -1045,19 +1051,106 @@ def writeMasterSolutionIndexCSV(outputPath, masterLibraries):
+  Raises:
+      ValueError: If ``inputArchs`` (i) contains a comma or space, (ii) refers to an unknown Gfx 
+      architecture, or (ii) malformed or use incorrect delimiters.
+
+  """
+  inputArchs = inputArchs.lower()
+
+  # CLI uses `;` delimiters, CMake uses `_` delimiters
+  archs = inputArchs.split(";" if ";" in inputArchs else "_")
+
+  output = set()
+  for arch in archs:
+    if arch == "all":
+      return set([arch])
+    elif arch in output:
+      printWarning(f"Removing duplicate architecture {arch}.")
+      continue
+    elif arch == "":  # Omit empty archs caused by leading and trailing delimiters
+      continue
+    elif arch not in architectureMap:
+      raise ValueError(f"`{arch}` is not a known Gfx architecture.")
+    else:
+      output.add(arch)
+
+  if len(output) == 0:
+    raise ValueError("Could not parse a viable target architecture from {inputArchs}. When specifying multiple options, "
+                        "use quoted, semicolon delimited architectures, e.g., `--architecture='gfx908;gfx1012'`")
+  return output
+
+def mapGfxArchitectures(gfxArchs: Set[str]) -> Set[str]:
+  """Validate and map target architectures from their Gfx name, to their for library creation.
+
+  Args:
+      Parsed input architectures.
+
+  Returns:
+      Architecture names associated with provided gfx names. See ``Common.architectureMap`` for mappings.
+
+  Raises:
+      ValueError: If an unsupported architecture is found.
+  """
+  if not isinstance(gfxArchs, set):
+    raise TypeError("`gfxArch` must be of type builtins.set")
+
+  namedArchs = set()
+  for arch in gfxArchs:
+    if arch in architectureMap:
+      namedArchs.add(architectureMap[arch])
+    else:
+      raise ValueError(f"`{arch}` is not a supported architecture.")
+  return namedArchs
+
+def findLogicFiles(path: Path, logicArchs: Set[str], lazyLoading: bool, experimentalDir: str, extraMatchers: Set[str]={"hip"}) -> List[str]:
+    """Recursively searches the provided path for logic files.
+
+    Args:
+        path: The path to the directory to search.
+        logicArchs: Target logic archiectures. These are interepreted as filename substrings
+            for which logic files are to be included.
+        extraMatchers: Additional directories to include for logic files.
+
+    Returns:
+        A list of Path objects representing the found YAML files.
+    """
+    isMatch = lambda file: any((arch in file.stem for arch in logicArchs.union(extraMatchers)))
+    isExperimental = lambda path: not experimentalDir in str(path)
+
+    extensions = ["*.yaml", "*.yml"]
+    logicFiles = filter(isMatch, (file for ext in extensions for file in path.rglob(ext)))
+    if not lazyLoading:
+      if not experimentalDir:
+        printWarning("Configuration parameter `ExperimentalLogicDir` is an empty string, "\
+                     "logic files may be filtered incorrectly.")
+      logicFiles = filter(isExperimental, logicFiles)
+
+    return list(str(l) for l in logicFiles)
```

Comment:
Are we converting to str because the downstream code is not path aware?

Reply:
Exactly.